Automattic/wp-calypso
# Social Auth: Use more precise Tracks events in Google Auth flows

Fixes #65649

## Proposed Changes

Introduces more precise Tracks events in Google Auth flows.

### Component Changes

`client/components/social-buttons/google.js` (the underlying functional component for connecting to Google)
* Renames `GoogleLoginButton` to `GoogleSocialButton` to reflect its more generic nature.
* Renames `calypso_login_social_auth_code_exchange_failure` event to `calypso_social_button_auth_code_exchange_failure` and adds `starting_point` prop. 997-gh-Automattic/tracks-events-registration
* Adds `calypso_social_button_auth_code_exchange_success` event with `social_account_type` and `starting_point` props. 998-gh-Automattic/tracks-events-registration
* Renames `calypso_login_social_button_failure` event to `calypso_social_button_failure` and adds `starting_point` prop. 1000-gh-Automattic/tracks-events-registration

### Signup Changes

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/36432/180458184-68fe6887-e243-489f-92dc-78b121a95aff.png">

`client/blocks/signup-form/social.jsx`
* Updates `GoogleLoginButton` to `GoogleSocialButton` and passes `startingPoint={ 'signup' }`.
* Renames `calypso_login_social_button_click` event to `calypso_signup_social_button_click` 995-gh-Automattic/tracks-events-registration
* Adds `calypso_signup_social_button_success` event with `starting_point` prop. 996-gh-Automattic/tracks-events-registration

### Login Changes

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/36432/180464301-91099d7a-e17c-41d8-a9a1-ee92928576ef.png">


`client/blocks/login/social.jsx`
* Updates `GoogleLoginButton` to `GoogleSocialButton` and passes `startingPoint={ 'signup' }`.

### Account Social Connect Changes

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/36432/180447049-a03edcc5-a412-466b-852b-35b4d2b26ae9.png">

`client/me/social-login/action-button.jsx`
* Adds `calypso_account_social_connect_button_click` event with `social_account_type` prop. 989-gh-Automattic/tracks-events-registration and 990-gh-Automattic/tracks-events-registration
* Adds `calypso_account_social_connect_success` event with `social_account_type` prop. 991-gh-Automattic/tracks-events-registration
* Adds `calypso_account_social_disconnect_button_click` event with `social_account_type` prop. 992-gh-Automattic/tracks-events-registration
* Adds `calypso_account_social_disconnect_success` event with `social_account_type` prop. 994-gh-Automattic/tracks-events-registration
* Adds `calypso_account_social_disconnect_failure` event with `error_code` and `social_account_type` prop. 993-gh-Automattic/tracks-events-registration

## Testing Instructions

See "Testing Instructions" in #64957 to set up a working `wpcalypso.wordpress.com` environment on your local machine. Hopefully it will be easier in the future: p1658341128755219-slack-C4EUDHL80

### Testing Signup

https://user-images.githubusercontent.com/36432/180463055-32651e7f-3e5c-40b9-9a55-9e935783ea41.mp4

In your `wpcalypso.wordpress.com` local environment:

1. Make sure you're logged out.
2. Navigate to `/start/user`
3. Run `localStorage.setItem( 'debug', 'calypso:analytics*' );` in your console.
4. Refresh the page for the debug to take effect.
5. Click on "Continue with Google" button.
6. Observe in your console: `Record event "calypso_signup_social_button_click" called with props {social_account_type: 'google'}`
7. Successfully complete the Google connect flow.
8. Observe in your console: `Record event "calypso_social_button_auth_code_exchange_success" called with props {social_account_type: 'google', starting_point: 'signup'}`
9. Observe in your console: `Record event "calypso_signup_social_button_success" called with props {social_account_type: 'google'}`

### Testing Login

https://user-images.githubusercontent.com/36432/180463857-fa1e44d5-f06c-4ea0-a36b-816d4d160b33.mp4

In your `wpcalypso.wordpress.com` local environment:

1. Make sure you're logged out.
2. Navigate to `/log-in`
3. Run `localStorage.setItem( 'debug', 'calypso:analytics*' );` in your console.
4. Refresh the page for the debug to take effect.
5. Click on "Continue with Google" button.
6. Observe in your console: `Record event "calypso_login_social_button_click" called with props {social_account_type: 'google'}`
7. Successfully complete the Google connect flow.
8. Observe in your console: `Record event "calypso_social_button_auth_code_exchange_success" called with props {social_account_type: 'google', starting_point: 'login'}`
9. Observe in your console: `Record event "calypso_login_social_login_success" called with props {social_account_type: 'google'}`

### Testing Account Social Connect

<img width="1558" alt="image" src="https://user-images.githubusercontent.com/36432/180446495-7df1c3b8-8a41-4a3a-99af-56793e735cf7.png">

In your `wpcalypso.wordpress.com` local environment:

1. Navigate to `/me/security/social-login`
2. Run `localStorage.setItem( 'debug', 'calypso:analytics*' );` in your console.
3. Refresh the page for the debug to take effect.
4. Click on Google's "Connect" button.
11. Observe in your console: `Record event "calypso_account_social_connect_button_click" called with props {social_account_type: 'google'}`
12. Successfully complete the Google connect flow.
13. Observe in your console: `calypso:analytics Record event "calypso_social_button_auth_code_exchange_success" called with props {social_account_type: 'google', starting_point: 'account-social-connect'}`
14. Observe in your console: `Record event "calypso_account_social_connect_success" called with props {social_account_type: 'google'}`
15. Click on Google's "Disconnect" button.
16. Observe in your console: `Record event "calypso_account_social_disconnect_button_click" called with props {social_account_type: 'google'}`
17. Observe in your console: `Record event "calypso_account_social_disconnect_success" called with props {social_account_type: 'google'}`
